### PR TITLE
fix(masthead): return focus to trigger items on close/escape

### LINKED ISSUE #8080: [Masthead]: React visual focus lost when closing a menu
**Engineering info:**  
- Innovation Team engineer: @andy-blum 
- Innovation Team JIRA Ticket: https://jsw.ibm.com/browse/HC-2373
- DPO consulting engineer: N/A

----

### Description

When the user presses the escape key to close a menu item the visual focus is not restored to the menu items in the masthead. User must tab once and that will bring them to the next menu item

https://user-images.githubusercontent.com/45692622/150207990-2f556596-cd49-48fc-a509-074062309121.mp4

### Component(s) impacted

All masthead examples

### Browser

Chrome

### Carbon for IBM.com version

1.30 rc 0

### Severity

Severity 2 = Aspects of design is broken, and impedes users in a significant way, but there is a way to complete their tasks. Affects major functionality, has a workaround.

### Application/website

https://carbon-design-system.github.io/carbon-for-ibm-dotcom/staging/react/?path=/story/components-masthead--default

### Package

@carbon/ibmdotcom-react

### CodeSandbox example

https://carbon-design-system.github.io/carbon-for-ibm-dotcom/staging/react/?path=/story/components-masthead--default

### Steps to reproduce the issue (if applicable)

- tab to menu
- enter to open menu
- tab through a few links
- press escape key to close menu

### Release date (if applicable)

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

### PR DESCRIPTION
**Pull request by andy-blum**
### Related Ticket(s)

Fixes #8080 #8081 

### Description

Implements a11y fix where focus is returned to the triggering element when dropdown menus/search form are closed

### Changelog

**Changed**

- focus-related accessibility fixes

### CODE REVIEW

**Review by emyarod — COMMENTED** [review]
looks good to me! just a couple thoughts

File: packages/react/src/components/Masthead/Masthead.js [hunk 1]
```diff
@@ -79,8 +79,13 @@ const Masthead = ({
    * @returns {*} The active search status
    */
   const [isSearchActive, setIsSearchActive] = useState(searchOpenOnload);
+  const searchIconButton = useRef(null);
+
   const handleChangeSearchActive = useCallback((event, { isOpen }) => {
     setIsSearchActive(isOpen);
+    setTimeout(() => {
+      searchIconButton.current.focus();
+    }, 0);
```
[the remark this answers is not in the document]
Comment (andy-blum) [reply packages/react/src/components/Masthead/Masthead.js:88]: Thanks for pointing this out - it's definitely redundant (as well as not working). I'll remove it!

The reason this doesn't work is that it sets focus on the search icon button, but during the `handleChangeSearchActive()` function that's called on the next line, the Masthead component re-renders everything in the masthead. Any component that _had_ focus, is removed from the DOM and focus is then delegated to the document body.

The timeout pushes the `.focus()` command to the back of the event loop thereby preventing focus until after the component is re-rendered and added back into the DOM.

[the remark this answers is not in the document]
Comment (emyarod) [reply packages/react/src/components/Masthead/Masthead.js:88]: right yeah, I thought there might be a way to eliminate the timeout and use a callback ref to deal with the changing ref value, but the only issue I kept running into was that it was setting focus on the button during the first render. thanks!

**Review by emyarod — APPROVED** [review]

**Review by IgnacioBecerra — APPROVED** [review]
LGTM, works as intended. Thanks @andy-blum!

**Pull request merged**